include/experimental/__p1684_bits/mdarray.hpp

```diff
@@ -20,8 +20,8 @@
 #include <cassert>
 #include <vector>
 
-namespace std {
-namespace experimental {
+namespace MDSPAN_IMPL_STANDARD_NAMESPACE {
+namespace MDSPAN_IMPL_PROPOSED_NAMESPACE {
```

Comment:
This forces mdarray to live in a nested namespace (e.g., `std::experimental`).  Users have no way to make it a single namespace without adding a `using` declaration or changing the code.

Should we consider adding a macro specific to mdarray, for the namespace(s) in which it lives?

Reply:
Would it be enough to have some sort of toggle, like `MDSPAN_IMPL_PROPOSED_NAMESPACE_IS_NESTED`? The problem with a combined macro is that it won't work in C++14 mode.

Reply:
@nmm0 wrote:
> Would it be enough to have some sort of toggle, like `MDSPAN_IMPL_PROPOSED_NAMESPACE_IS_NESTED`?

The `mdarray` implementation currently assumes that it's living inside `MDSPAN_IMPL_STANDARD_NAMESPACE` and therefore doesn't need to namespace-qualify `mdspan` components like `extents`.  Fixing that, and testing the fix, would be a lot more effort.

> The problem with a combined macro is that it won't work in C++14 mode.

True, then you would need separate macros for "the name of the possibly nested namespace" (e.g., `std::experimental`), "open a scope in the existing namespace" (e.g., `namespace std { namespace experimental {`), and "close a scope in the existing namespace" (e.g., `}}`).

Reply:
I am open to dropping support for 14 and then have MDSPAN_IMPL_

> I also had to add an extra define `MDSPAN_IMPL_NAMESPACE_STD` when we are in "std-like" mode because of conflicts with `<span>`

 Can't we fix that with a conditional include exits thingy?

Reply:
Or even just we don't define our own dynamic_extents when in C++20 mode ...

Reply:
I update to import `dynamic_extents` into the mdspan namespace.